src/test/run-pass/traits/traits-default-method-macro2.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2012 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+//
+// run-pass
+//
+// Description - allow a method declared in a macro to act as a
+// default trait implementation.
+
+macro_rules! def_fn {
+    ($body:block) => {
+        fn bar() $body
+    }
+}
+
+trait Foo {
+    def_fn!({ println!("bar"); });
+}
```

Comment:
Thanks =) I think it'd be good to include in the text the other various places one might use this as well -- e.g., for a free function etc. Skimming the existing tests I did find that `macros/macro-interpolation.rs` and `macros/macro-stmt` both test the free fn case but there don't appear to be tests in impls.

Reply:
I think it'd be good to just have one single test dedicated to testing that `$b` can be used as a fn body in all relevant contexts (the other tests are kind of testing a wider variety of things)

Reply:
Noted. will update and let you know when its ready